### PR TITLE
[processing] Don't use crappy Qt file picker dialog

### PR DESCRIPTION
**Pull request by nyalldawson**
Moves the output destination encoding option to a separate menu action instead of embedding it in the output destination file selector. Otherwise Qt downgrades the file picker to the Qt basic file picker, instead of the native OS file picker. Which is a huge downgrade and barely functional for users.

I watched a user struggle with this limited dialog last week - they couldn't find any of their network paths and the desktop shortcuts weren't working, so they just gave up an saved the output on their desktop. I don't blame them - I blame the Qt dialog.

Also remember last selected file format.

### CODE REVIEW

**Comment by nirvn** [comment]
Woupidou!

**Comment by DelazJ** [comment]
I just experienced the issue and was about to check the issue report!!!

**Comment by nirvn** [comment]
I only have one request for change on this PR: you need to edit the commit message and find a stronger adjective, "crappy" doesn't even begin to describe it! :wink:

**Comment by elpaso** [comment]
:+1: for adding "crappy" to the spell checker whitelist! :wink:

File: python/plugins/processing/gui/DestinationSelectionPanel.py [hunk 1]
```diff
@@ -33,12 +33,13 @@
 from qgis.PyQt.QtCore import QCoreApplication, QDir
 from qgis.PyQt.QtWidgets import QDialog, QMenu, QAction, QFileDialog
 from qgis.PyQt.QtGui import QCursor
-from qgis.gui import QgsEncodingFileDialog, QgsExpressionBuilderDialog
+from qgis.gui import QgsEncodingFileDialog, QgsEncodingSelectionDialog
```
Comment (alexbruy) [inline python/plugins/processing/gui/DestinationSelectionPanel.py]: Looks like `QgsEncodingFileDialog` can be safely removed now

**Review by alexbruy — APPROVED** [review]
Looks good to me. Death to crappy dialogs

**Pull request merged**